projects/Magnetosphere/Magnetosphere.cpp

```diff
@@ -390,8 +408,11 @@ namespace projects {
       
 #pragma omp parallel
       {
+         bool doZeroOut;
          //Force field to zero in the perpendicular direction for 2D (1D) simulations. Otherwise we have unphysical components.
-         if(P::xcells_ini==1) {
+         doZeroOut = P::xcells_ini ==1 && this->zeroOutComponents[0]==1;
+      
+         if(doZeroOut) {
```

Comment:
I would prefer this in one `if(P::xcells_ini ==1 && this->zeroOutComponents[0]==1)`

Reply:
Done